app/FirebaseApi.js

```diff
@@ -0,0 +1,46 @@
+/* global someFunction window:true */
+
+import firebase from 'firebase';
+import RNFetchBlob from 'react-native-fetch-blob';
+import Config from 'react-native-config';
+
+// Init Firebase
+const config = {
+  apiKey: Config.FIREBASE_API_KEY,
+  authDomain: 'freemarketrn.firebaseapp.com',
+  storageBucket: 'freemarketrn.appspot.com',
+};
+firebase.initializeApp(config);
+const storage = firebase.storage();
+
+// Prepare Blob support
+const Blob = RNFetchBlob.polyfill.Blob;
+const fs = RNFetchBlob.fs;
+window.XMLHttpRequest = RNFetchBlob.polyfill.XMLHttpRequest;
+window.Blob = Blob;
```

Comment:
i don't know for sure.
I read that it is defined by React Native. 
I suppose we redefine here built-in XMLHttpRequest and Blob. And they are called later through window. At least, i doesn't work without it.